metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -388,7 +389,12 @@ public void replay(RemoveTopicRecord record) {
         Map<String, CreatableTopicResult> successes = new HashMap<>();
         for (CreatableTopic topic : request.topics()) {
             if (topicErrors.containsKey(topic.name())) continue;
-            ApiError error = createTopic(topic, records, successes);
+            ApiError error;
+            try {
+                error = createTopic(topic, records, successes);
```

Comment:
I find it strange that `createTopic` can return an `ApiError` or throw an `ApiException`. I think it should do one or the other but not both. What is now throwing that requires a `try ... catch ...`?

Reply:
The try...catch is needed for cases where an unexpected exception is thrown. I agree that this could be refactored to just be exception-based. But let's do that in a follow-on since this change is already big enough...